api/src/organizational-entities/domain/usecases/archive-organizations-in-batch.usecase.js

```diff
@@ -0,0 +1,33 @@
+import { ArchiveOrganizationsInBatchError } from '../errors.js';
+
+/**
+ * @param {Object} params
+ * @param {Array} params.organizationIds
+ * @param {Number} params.userId
+ * @param {Object} params.organizationForAdminRepository
+ */
+export const archiveOrganizationsInBatch = async function ({
+  organizationIds,
+  userId,
+  organizationForAdminRepository,
+}) {
+  // we don't use a transaction here not to rollback lines 0 to N-1 in case of an error on line N
+
+  for (const [index, organizationId] of organizationIds.entries()) {
+    const hasOrganizationBeenArchived = Boolean(
+      await organizationForAdminRepository.archive({
+        id: organizationId,
+        archivedBy: userId,
+      }),
+    );
+
+    if (!hasOrganizationBeenArchived) {
+      throw new ArchiveOrganizationsInBatchError({
+        meta: {
+          currentLine: index + 1,
+          totalLines: organizationIds.length,
+        },
+      });
+    }
+  }
```

Comment:
Pourquoi ne pas utiliser un `try`/`catch` dans le `for...of` comme cela est fait dans `api/src/organizational-entities/domain/usecases/archive-certification-centers-in-batch.usecase.js` ?

Reply:
Voir commentaire du dessus 😉